rmf_robot_sim_gz_plugins/src/TeleportDispenser.cpp

```diff
@@ -65,19 +69,42 @@ class GZ_SIM_VISIBLE TeleportDispenserPlugin
   void PreUpdate(const UpdateInfo& info, EntityComponentManager& ecm) override;
 
 private:
-  // Stores params representing state of Dispenser, and handles the main dispenser logic
-  std::unique_ptr<TeleportDispenserCommon> _dispenser_common;
-
   gz::transport::Node _ign_node;
 
   Entity _dispenser;
   Entity _item_en; // Item that dispenser may contain
   gz::math::AxisAlignedBox _dispenser_vicinity_box;
 
-  bool tried_fill_dispenser = false; // Set to true if fill_dispenser() has been called at least once
+  bool _dispense = false;
+  DispenserRequest _latest; // Only store and act on last received request
+
+  std::string _guid; // Plugin name
 
+  double _last_pub_time = 0.0;
+  double _sim_time = 0.0;
+
+  bool _item_en_found = false; // True if entity to be dispensed has been determined. Used when locating item in future
+  bool _dispenser_filled = false;
+
+  std::unordered_map<std::string, FleetState::UniquePtr> _fleet_states;
+  DispenserState _current_state;
   rclcpp::Node::SharedPtr _ros_node;
 
+  bool tried_fill_dispenser = false; // Set to true if fill_dispenser() has been called at least once
+
+  rclcpp::Subscription<FleetState>::SharedPtr _fleet_state_sub;
+  rclcpp::Publisher<DispenserState>::SharedPtr _state_pub;
+  rclcpp::Subscription<DispenserRequest>::SharedPtr _request_sub;
+  rclcpp::Publisher<DispenserResult>::SharedPtr _result_pub;
+  std::unordered_map<std::string, bool> _past_request_guids;
+
+  void init_ros_node();
+  void send_dispenser_response(uint8_t status) const;
+  void fleet_state_cb(FleetState::UniquePtr msg);
+  void dispenser_request_cb(DispenserRequest::UniquePtr msg);
+  void on_update(EntityComponentManager& ecm);
+
+  void try_refill_dispenser(EntityComponentManager& ecm);
   SimEntity find_nearest_model(
```

Comment:
We can get rid of the concept of `SimEntity` and just use `Entity`, this was created to have a "generic" concept for both Gazebo and Gazebo Classic but now only makes things harder to parse